direct/core/src/main/java/cz/o2/proxima/direct/transaction/ClientTransactionManager.java

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright 2017-2021 O2 Czech Republic, a.s.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package cz.o2.proxima.direct.transaction;
+
+import cz.o2.proxima.functional.BiConsumer;
+import cz.o2.proxima.transaction.KeyAttribute;
+import cz.o2.proxima.transaction.Response;
+import java.util.List;
+
+public interface ClientTransactionManager extends AutoCloseable, TransactionManager {
```

Comment:
Isn't `pre-commit` method missing here? This is imported to support exactly-once in distributed environments (eg. two phase commit in flink)

Reply:
The pre-commit is actually the commit in this sense. The client is still responsible for writing the data - this needs polishing, because we must ensure enventual consistency in case when there are multiple updates in transaction. That will be addressed in next PR(s).